planner/core/rule_eliminate_projection.go

```diff
@@ -70,6 +70,7 @@ func resolveExprAndReplace(origin expression.Expression, replace map[string]*exp
 	}
 }
 
+//may be we can spare the idle physicalPlan into free pool rather than throwing it to gc
```

Comment:
Such comment is not appropriate as the comment of function header.
You can open an issue and then discuss with us.